diff --git a/libs/glib2/Makefile b/libs/glib2/Makefile
index 583828b..4e46165 100644
--- a/libs/glib2/Makefile
+++ b/libs/glib2/Makefile
@@ -46,7 +46,7 @@ endef
 
 HOST_CONFIGURE_ARGS += \
 	--disable-selinux \
-	--with-libiconv=gnu
+	$(if $(ICONV_FULL),--with-libiconv=gnu)
 
 CONFIGURE_ARGS += \
 	--enable-shared \
@@ -54,7 +54,7 @@ CONFIGURE_ARGS += \
 	--enable-debug=no \
 	--disable-selinux \
 	--disable-fam \
-	--with-libiconv=gnu
+	$(if $(ICONV_FULL),--with-libiconv=gnu)
 
 CONFIGURE_VARS += \
 	glib_cv_stack_grows=no \
