diff --git a/lang/node/Makefile b/lang/node/Makefile
index 7effb08..ee3c0c9 100644
--- a/lang/node/Makefile
+++ b/lang/node/Makefile
@@ -65,6 +65,10 @@ HOST_CONFIGURE_ARGS:= \

 HOST_CONFIGURE_CMD:=python ./configure

+define Build/Compile
+    $(call Build/Compile/Default,all V=)
+endef
+
 define Build/InstallDev
        $(INSTALL_DIR) $(1)/usr/include
        $(CP) $(PKG_INSTALL_DIR)/usr/include/* $(1)/usr/include/

